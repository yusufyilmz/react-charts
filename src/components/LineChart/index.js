
import * as React from "react";
import ChartPath from '../ChartPath';

class LineChart extends React.Component {
    render() {
        let { width, height, color } = this.props
        return (
            <svg width={width} x="-10.5em" y="1.5em" height={height}>
                <ChartPath
                    {...this.props}
                    radius={130}
                    strokeColor={color}
                    color={"none"}
                    type="line"
                />
            </svg>
        )
    }
}

export default LineChart;
