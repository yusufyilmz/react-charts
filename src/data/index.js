
export default [
    {
        title: 'REVENUE',
        totalValue: 200000,
        valueType: 'euro',
        values: [
            {
                name: 'Tablet',
                percentage: 60,
                color: '#47d147'
            },
            {
                name: 'Smartphone',
                percentage: 40,
                color: '#006600'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
    {
        title: 'IMPRESSIONS',
        totalValue: 200000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 40,
                color: '#1ad1ff'
            },
            {
                name: 'Smartphone',
                percentage: 60,
                color: '#002699'
            }
        ],
        datas: [
            { id: 1, value: 5.2 },
            { id: 2, value: 4.8 },
            { id: 3, value: 5.0 },
            { id: 4, value: 3.8 },
            { id: 5, value: 2.3 },
            { id: 6, value: 2.8 },
            { id: 7, value: 1.5 },
            { id: 8, value: 2.5 },
            { id: 9, value: 2.0 },
        ]
    },
    {
        title: 'VISITS',
        totalValue: 600000000,
        valueType: '',
        values: [
            {
                name: 'Tablet',
                percentage: 80,
                color: '#ffcc00'
            },
            {
                name: 'Smartphone',
                percentage: 20,
                color: '#e68a00'
            }
        ],
        datas: [
            { id: 1, value: 1.1 },
            { id: 2, value: 2.2 },
            { id: 3, value: 1.8 },
            { id: 4, value: 3.2 },
            { id: 5, value: 2.5 },
            { id: 6, value: 5.2 },
            { id: 7, value: 4.6 },
            { id: 8, value: 6.3 },
            { id: 9, value: 4.8 },
        ]
    },
]

