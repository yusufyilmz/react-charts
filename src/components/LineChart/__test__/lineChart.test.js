

import React from 'react';
import LineChart from '../index';
import { shallow } from 'enzyme'
import data from '../../../data';

describe('LINECHART COMPONENT', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            data: data[0].datas, 
            width: 400,
            height: 200,
            margin: 1,
            color: 'black'
        }

        wrapper = shallow(
            <LineChart {...props} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find('svg').length).toBe(1)
    });

    test('renders props correctly', () => {
        expect(wrapper.find('svg').props()['width']).toBe(props.width)
    });

    test('renders props correctly', () => {
        expect(wrapper.find('svg').props()['height']).toBe(props.height)
    });

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})
