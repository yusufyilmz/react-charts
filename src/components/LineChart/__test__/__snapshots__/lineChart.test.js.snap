// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LINECHART COMPONENT snapchat correctly 1`] = `
<svg
  height={200}
  width={400}
  x="-10.5em"
  y="1.5em"
>
  <ChartPath
    color="none"
    data={
      Array [
        Object {
          "id": 1,
          "value": 1.1,
        },
        Object {
          "id": 2,
          "value": 2.2,
        },
        Object {
          "id": 3,
          "value": 1.8,
        },
        Object {
          "id": 4,
          "value": 3.2,
        },
        Object {
          "id": 5,
          "value": 2.5,
        },
        Object {
          "id": 6,
          "value": 5.2,
        },
        Object {
          "id": 7,
          "value": 4.6,
        },
        Object {
          "id": 8,
          "value": 6.3,
        },
        Object {
          "id": 9,
          "value": 4.8,
        },
      ]
    }
    height={200}
    margin={1}
    radius={130}
    strokeColor="black"
    type="line"
    width={400}
  />
</svg>
`;
